pkg/filesystem/subvolume.go

```diff
@@ -53,7 +53,9 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	subvolumeNames := make(map[string]subVolumeInfo)
 	for _, pv := range pvList.Items {
 		if pv.Spec.CSI != nil && pv.Spec.CSI.VolumeAttributes["subvolumeName"] != "" {
-			subvolumeNames[pv.Spec.CSI.VolumeAttributes["subvolumeName"]] = subVolumeInfo{}
+			if pv.Spec.CSI.VolumeAttributes["subvolumeName"] != "" {
```

Comment:
in which case the subvolumeName can be empty?

Reply:
It was added as a check when there will be no CSI subvolumes, but yeah with your last PR to fix this issue, got to understand the correct check.
if there is a CSI volume, don't think the subvolumename will be empty. I will remove the check, it's not required.